modules/core/src/lib/layer-manager.js

```diff
@@ -88,7 +88,7 @@ export default class LayerManager {
 
       gl,
       // Enabling luma.gl Program caching using private API (_cachePrograms)
-      shaderCache: new ShaderCache({gl, _cachePrograms: true}),
+      shaderCache: gl && new ShaderCache({gl, _cachePrograms: true}),
```

Comment:
If we are going to make certain classes work without WebGL, to make sure we don't regress, we should ideally at least have test cases that instantiate them without WebGL context.